src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -448,6 +475,7 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                     method.IsStatic &&
                     method.ParameterCount == 0 &&
                     !method.IsGenericMethod &&
+                    method.RefKind == RefKind.None &&
```

Comment:
Is this actually just a bug fix?

Reply:
We need this change for proper behavior of method-level builders (see test `BuilderOnMethod_CreateHasRefReturn`).
But yes it was a misbehavior of task-like types (see test `AsyncMethod_CreateHasRefReturn`).